controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -218,35 +220,16 @@ func (r *AutoscalingRunnerSetReconciler) Reconcile(ctx context.Context, req ctrl
 		log.Info("Find existing ephemeral runner set", "name", runnerSet.Name, "specHash", runnerSet.Labels[labelKeyRunnerSpecHash])
 	}
 
-	if desiredSpecHash != latestRunnerSet.Labels[labelKeyRunnerSpecHash] {
-		log.Info("Latest runner set spec hash does not match the current autoscaling runner set. Creating a new runner set")
-		return r.createEphemeralRunnerSet(ctx, autoscalingRunnerSet, log)
-	}
-
-	oldRunnerSets := existingRunnerSets.old()
-	if len(oldRunnerSets) > 0 {
-		log.Info("Cleanup old ephemeral runner sets", "count", len(oldRunnerSets))
-		err := r.deleteEphemeralRunnerSets(ctx, oldRunnerSets, log)
-		if err != nil {
-			log.Error(err, "Failed to clean up old runner sets")
-			return ctrl.Result{}, err
-		}
-	}
-
 	// Make sure the AutoscalingListener is up and running in the controller namespace
 	listener := new(v1alpha1.AutoscalingListener)
 	if err := r.Get(ctx, client.ObjectKey{Namespace: r.ControllerNamespace, Name: scaleSetListenerName(autoscalingRunnerSet)}, listener); err != nil {
 		if kerrors.IsNotFound(err) {
-			// We don't have a listener
-			log.Info("Creating a new AutoscalingListener for the runner set", "ephemeralRunnerSetName", latestRunnerSet.Name)
-			return r.createAutoScalingListenerForRunnerSet(ctx, autoscalingRunnerSet, latestRunnerSet, log)
+			log.Info("AutoscalingListener does not exist.")
```

Comment:
We should keep the state of not found in a variable instead of checking with `reflect.DeepEqual`
```go
	listener := new(v1alpha1.AutoscalingListener)
        listenerFound := true
	if err := r.Get(ctx, client.ObjectKey{Namespace: r.ControllerNamespace, Name: scaleSetListenerName(autoscalingRunnerSet)}, listener); err != nil {
		if kerrors.IsNotFound(err) {
			log.Info("AutoscalingListener does not exist.")
                        listenerFound = false
		}
	}
```
Then it becomes:
```go
if listenerFound && listener.Labels[labelKeyRunnerSpecHash] != autoscalingRunnerSet.ListenerSpecHash() {
// delete listener
}
if listenerFound && desiredSpecHash != latestRunnerSet.Labels[labelKeyRunnerSpecHash] {
    if r.drainingJobs(&latestRunnerSet.Status) {
    // patch
    }
    // create ephemeral runner set
}
// old runner set checks

if !listenerFound  {
    if r.drainingJobs(&latestRunnerSet.Status) {
           // log and return
    }
    // create 
}
// update status
```

Reply:
Smart! I like it much more efficient

Reply:
Fixed